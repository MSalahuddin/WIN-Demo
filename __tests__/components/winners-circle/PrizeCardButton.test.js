import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import PrizeCardButton from '../../../js/components/winners-circle/PrizeCardButton';
import MockProvider from '../../MockProvider';

describe('Component PrizeCardButton', () => {
  const onPress = jest.fn();

  test('renders regular button correctly', () => {
    const output = render(
      <MockProvider>
        <PrizeCardButton value={10} onPress={onPress} newValue={null} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders discounted prize button correctly', () => {
    const output = render(
      <MockProvider>
        <PrizeCardButton value={10} onPress={onPress} newValue={5} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when big redeem button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <PrizeCardButton value={10} onPress={onPress} newValue={5} />
      </MockProvider>
    );
    fireEvent(getByTestId('redeem-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });

  test('calls onPress when small discounted price button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <PrizeCardButton value={10} onPress={onPress} newValue={5} />
      </MockProvider>
    );
    fireEvent(getByTestId('discounted-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
