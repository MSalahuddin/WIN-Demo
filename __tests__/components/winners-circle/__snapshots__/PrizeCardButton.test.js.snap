// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrizeCardButton renders discounted prize button correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(160,160,160,1)",
        "borderRadius": 106,
        "height": 96,
        "justifyContent": "center",
        "opacity": 1,
        "width": 278,
      }
    }
    testID="redeem-button"
  >
    <View
      borderRadius={53}
      colorScheme="rgba(209,209,209,1)"
      height={48}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(209,209,209,1)",
            "borderRadius": 106,
            "height": 86,
            "justifyContent": "center",
            "width": 272,
          },
        ]
      }
      width={139}
    >
      <View
        borderRadius={53}
        colorScheme="rgba(183,183,183,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(183,183,183,1)",
              "borderRadius": 126,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 8,
              "width": 272,
            },
          ]
        }
        width={139}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            size={29}
            source={1}
            style={
              Array [
                Object {
                  "height": 43.5,
                  "marginRight": 16,
                  "width": 43.5,
                },
              ]
            }
          />
          <View>
            <Text
              alignCenter={false}
              allowFontScaling={true}
              color="#E5E5E5"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              marginTop={8}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#E5E5E5",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "left",
                    "textDecorationColor": "#E5E5E5",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginTop": 16,
                  },
                ]
              }
            >
              10
            </Text>
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 3,
                    "borderColor": "#F06B59",
                    "height": 1,
                    "position": "absolute",
                    "top": "50%",
                    "width": "100%",
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "position": "absolute",
                "right": -7.5,
                "top": -44,
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(197,78,62,1)",
                  "borderRadius": 22,
                  "height": 44,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 122,
                }
              }
              testID="discounted-button"
            >
              <View
                borderRadius={11}
                colorScheme="rgba(255,132,115,1)"
                height={22}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(255,132,115,1)",
                      "borderRadius": 22,
                      "height": 34,
                      "justifyContent": "center",
                      "width": 116,
                    },
                  ]
                }
                width={61}
              >
                <View
                  borderRadius={11}
                  colorScheme="rgba(240, 107, 89, 1)"
                  height={22}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(240, 107, 89, 1)",
                        "borderRadius": 42,
                        "height": 30,
                        "justifyContent": "center",
                        "marginTop": 8,
                        "width": 116,
                      },
                    ]
                  }
                  width={61}
                >
                  <Text
                    alignCenter={false}
                    allowFontScaling={true}
                    color="#FFFFFF"
                    fontFamily="BOLD ITALIC"
                    isUnderlined={false}
                    isUppercase={false}
                    marginTop={0}
                    maxFontSizeMultiplier={1.3}
                    size="SMALL"
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontFamily": "CalibreBoldItalic",
                          "fontSize": 27,
                          "justifyContent": "center",
                          "textAlign": "left",
                          "textDecorationColor": "#FFFFFF",
                          "textDecorationLine": "none",
                          "textDecorationStyle": "solid",
                          "textTransform": "none",
                        },
                        Object {
                          "marginTop": 0,
                        },
                      ]
                    }
                  >
                    5
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component PrizeCardButton renders regular button correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(160,160,160,1)",
        "borderRadius": 106,
        "height": 96,
        "justifyContent": "center",
        "opacity": 1,
        "width": 278,
      }
    }
    testID="redeem-button"
  >
    <View
      borderRadius={53}
      colorScheme="rgba(209,209,209,1)"
      height={48}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(209,209,209,1)",
            "borderRadius": 106,
            "height": 86,
            "justifyContent": "center",
            "width": 272,
          },
        ]
      }
      width={139}
    >
      <View
        borderRadius={53}
        colorScheme="rgba(183,183,183,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(183,183,183,1)",
              "borderRadius": 126,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 8,
              "width": 272,
            },
          ]
        }
        width={139}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            size={29}
            source={1}
            style={
              Array [
                Object {
                  "height": 43.5,
                  "marginRight": 16,
                  "width": 43.5,
                },
              ]
            }
          />
          <View>
            <Text
              alignCenter={false}
              allowFontScaling={true}
              color="#E5E5E5"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              marginTop={8}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#E5E5E5",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "left",
                    "textDecorationColor": "#E5E5E5",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginTop": 16,
                  },
                ]
              }
            >
              10
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
